config/routes.rb

```diff
@@ -3,4 +3,8 @@
 
   # Defines the root path route ("/")
   # root "articles#index"
+  get 'users', to: 'users#index'
+  get 'users/:id', to: 'users#show'
+  get 'users/:user_id/posts', to: 'posts#index'
+  get 'users/:user_id/posts/:id', to: 'posts#show'
```

Comment:
- [ ] [Optional] It would be better professionally if you create your routes like that:
```ruby
resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
end
```
- [ ] [Optional] Please make the user index your root route like that:
```ruby
root 'users#index'
```